packages/manager/apps/pci-object-storage/src/pages/objects/container/enable-encryption/EnableEncryption.page.tsx

```diff
@@ -0,0 +1,145 @@
+import { ApiError } from '@ovh-ux/manager-core-api';
+import { PciModal } from '@ovh-ux/manager-pci-common';
+import { useNotifications } from '@ovh-ux/manager-react-components';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import { OdsLink, OdsMessage, OdsText } from '@ovhcloud/ods-components/react';
+import { useContext } from 'react';
+import { Translation, useTranslation } from 'react-i18next';
+import {
+  createSearchParams,
+  useNavigate,
+  useParams,
+  useSearchParams,
+} from 'react-router-dom';
+import { PAGE_PREFIX } from '@/tracking.constants';
+import { useAllStorages, useUpdateStorage } from '@/api/hooks/useStorages';
+import { ENABLE_ENCRYPTION_LINK } from '@/constants';
+
+export default function EnableEncryptionPage() {
+  const { t } = useTranslation('containers/enable-encryption');
+
+  const { addSuccess, addError } = useNotifications();
+  const { tracking } = useContext(ShellContext).shell;
+  const navigate = useNavigate();
+  const { projectId, storageId } = useParams();
+
+  const [searchParams] = useSearchParams();
+
+  const { data: storages, isPending: isStoragesPending } = useAllStorages(
+    projectId,
+  );
+  const context = useContext(ShellContext);
+
+  const { ovhSubsidiary } = context.environment.getUser();
+
+  const storageDetail = storages?.resources.find((s) => s.name === storageId);
```

Comment:
useMemo or useQuery with select

Also, don't you have a query to get this specific storage instead ?